### PR TITLE
Modernize qlpacks and update to CodeQL 2.8.5

### PR DESCRIPTION
**Pull request by jketema**
## Description

(Note that this PR targets the `next` branch, which is used by the CI for CodeQL to detect changes that break `github/codeql-coding-standards`. These changes should eventually be merged to `main`, but that should not happen until we are ready to update to the required toolset.)

This PR updates the qlpacks and related files to use CodeQL packaging to declare and resolve dependencies. Neither the `codeql_modules` submodule nor the `codeql_home/codeql-stdlib` checkout of `github/codeql` are required anymore. The necessary dependencies can be installed by running `python3 scripts/install-packs.yml`. This need only be run when we update to a new standard library version, but it doesn't hurt to run it after every pull.

None of the `codeql` commands we run should require `--search-path` or `--additional-packs` any longer. References within our repo are resolved automatically because all of our packs are in the same "CodeQL Workspace" (defined by our `.codeqlmanifest.json` file. References to the standard library are resolved from the package download cache.

Details:
- Adds `codeql/` scope to all pack names, which is necessary to publish packs to the pack registry. Note that we don't actually publish to the package registry yet.
- Uses `dependencies:` map instead of `libraryPathDependencies:` list. The `'*'` version indicates that the referenced pack will be found in the local workspace, rather than downloaded from the registry.
- Folds the `.codeqlmanifest.json` in the `cpp` directory into the root `.codeqlmanifest.json`. All pack references should resolve automatically now, without a need for `--search-path`
- Adds `.codeql` directories to `.gitignore`. `.codeql` is where the output of building a query or library pack is placed.
- Removes `--search-path` and `--additional-packs` from all of our CI workflows.
- Adds a new composite Action to install the standard library dependencies.
- Adds a new workflow to validate that the actual standard library dependencies specified in the various `qlpack.yml` files match the versions in the `github/codeql` commit specified in `supported_codeql_configs.json`. The script used by this workflow will also be used, in a future PR, in the workflow to update to a new toolset version.
- Updates the supported toolset to 2.8.5. The 2.7.6 release that we were already on has a blocking bug in the C++ standard library pack.
- Fixes a test expectation file based on changes in the set of reported paths for one path-problem query due to changes in the standard dataflow library.
- Updates the developer handbook with the necessary information.

## Change request type

 - [x] Release or process automation (GitHub workflows, internal scripts)
 - [x] Internal documentation
 - [ ] External documentation
 - [ ] Query files (`.ql`, `.qll`, `.qls` or unit tests)
 - [ ] External scripts (analysis report or other code shipped as part of a release)

## Rules with added or modified queries

 - [ ] No rules added
 - [ ] Queries have been added for the following rules:
     - _rule number here_
 - [ ] Queries have been modified for the following rules:
     - _rule number here_

## Release change checklist

A change note ([development_handbook.md#change-notes](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#change-notes)) is required for any pull request which modifies:

 - The structure or layout of the release artifacts.
 - The evaluation performance (memory, execution time) of an existing query.
 - The results of an existing query in any circumstance.

If you are only adding new rule queries, a change note is not required.

_**Author:**_ Is a change note required? 
  - [ ] Yes
  - [x] No

_**Reviewer:**_ Confirm that either a change note is not required or the change note is required and has been added.
  - [ ] Confirmed

## Query development review checklist

For PRs that add new queries or modify existing queries, the following checklist should be completed by both the author and reviewer:

### Author

 - [ ] Have all the relevant rule package description files been checked in?
 - [ ] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [ ] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [ ] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [ ] Does the query have an appropriate level of in-query comments/documentation?
 - [ ] Have you considered/identified possible edge cases?
 - [ ] Does the query not reinvent features in the standard library?
 - [ ] Can the query be simplified further (not golfed!)

### Reviewer 

 - [ ] Have all the relevant rule package description files been checked in?
 - [ ] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [ ] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [ ] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [ ] Does the query have an appropriate level of in-query comments/documentation?
 - [ ] Have you considered/identified possible edge cases?
 - [ ] Does the query not reinvent features in the standard library?
 - [ ] Can the query be simplified further (not golfed!)

### CODE REVIEW

**Review by rvermeulen — CHANGES_REQUESTED** [review]
Looks good. We started to standardize our workflow and Python scripts on 3.9, so I made a few suggestion to make that explicit.

File: .github/actions/action.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  mode:
+    description: |
+      The `--mode` option to `codeql pack install`.
+    required: true
+    default: verify
+
+runs:
+  using: composite
+  steps:
+    - name: Install CodeQL library packs
```
[the remark this answers is not in the document]
Comment (jketema) [reply .github/actions/action.yml]: I've delated this file. I accidentally copied it twice, once in the wrong location (this one).

File: .github/workflows/verify-standard-library-dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      matrix: ${{fromJSON(needs.prepare-matrix.outputs.matrix)}}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python 3
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x"
```
[the remark this answers is not in the document]
Comment (jketema) [reply .github/workflows/verify-standard-library-dependencies.yml]: Done.

File: .github/actions/install-codeql-packs/action.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  mode:
+    description: |
+      The `--mode` option to `codeql pack install`.
+    required: true
+    default: verify
+
+runs:
+  using: composite
+  steps:
+    - name: Install CodeQL library packs
```
[the remark this answers is not in the document]
Comment (jketema) [reply .github/actions/install-codeql-packs/action.yml:19]: https://github.com/github/codeql-coding-standards/runs/7362437649?check_suite_focus=true#step:6:2

[the remark this answers is not in the document]
Comment (rvermeulen) [reply .github/actions/install-codeql-packs/action.yml:19]: I'm really bad at specifying YAML in the suggestion functionality, please make the change with your favourite editor that can handle YAML.

**Review by rvermeulen — APPROVED** [review]
LGTM

**Comment by rvermeulen** [comment]
The failing check has been made optional in `main`. So if you merge that back into `next` it should pass.

**Comment by jketema** [comment]
> The failing check has been made optional in main. So if you merge that back into jketema:modernize-packs it should pass.

Done.

**Comment by jketema** [comment]
> > The failing check has been made optional in main. So if you merge that back into jketema:modernize-packs it should pass.

> Done.

Slightly backtracked on this to keep the PR "clean" for prosperity. Since `next` was just behind `main` I simply moved `next` up to the head of `main` and rebased.

**Comment by jketema** [comment]
@rvermeulen Not sure if you want to re-approved because of the rebase?

**Pull request merged**